src/sys/aio.rs

```diff
@@ -239,6 +244,37 @@ impl<'a> AioCb<'a> {
         })
     }
 
+    /// Returns the `aiocb`'s `LioOpcode` field
+    ///
+    /// If the value cannot be represented as an `LioOpcode`, returns `None`
+    /// instead.
+    pub fn lio_opcode(&self) -> Option<LioOpcode> {
+        match self.aiocb.aio_lio_opcode {
+            libc::LIO_READ => Some(LioOpcode::LIO_READ),
+            libc::LIO_WRITE => Some(LioOpcode::LIO_WRITE),
+            libc::LIO_NOP => Some(LioOpcode::LIO_NOP),
+            _ => None
+        }
+    }
+
+    /// Returns the desired length of the aio operation in bytes
+    ///
+    /// If the operation has completed, this does *not* return the completed
```

Comment:
Can we embed this into the type system somehow?

Reply:
No, but I could probably word it better.